src/components/portfolio/DetailsMainAnalyticsHeatmap.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import Chart from 'react-apexcharts';
+import { useTheme } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import { NonEmptyPortfolioDetails } from '../../portfolio/APIClient';
+
+type MappedCorrelation = [string, string, number];
+
+type HeatMapSeries = {
+  name: string;
+  // the order of which number represents which company is always the same
+  data: number[];
+};
+
+type HeatmapProps = {
+  portfolio: NonEmptyPortfolioDetails;
+  height: number;
+};
+
+const DetailsAnalyticsHeatmap: React.FC<HeatmapProps> = ({
+  portfolio,
+  height,
+}) => {
+  const theme = useTheme();
+  const { t } = useTranslation();
+
+  const { correlations } = portfolio.analytics;
+
+  // array of arrays of symbol - symbol - correlation
+  const mappedCorrelations: MappedCorrelation[] = Object.keys(
+    correlations
+  ).map((key) => [key.split(';')[0], key.split(';')[1], correlations[key]]);
+
+  // gets unique set of companies displayed in the chart (used as x-axis categories)
+  const chartCategories = Array.from(
+    new Set(
+      mappedCorrelations
+        .map((mc) => [mc[0], mc[1]])
+        .reduce((arr1, arr2) => arr1.concat(arr2))
+        .sort()
+    )
+  );
+
+  // will be passed to the chart as series prop
+  const finalSeries: HeatMapSeries[] = [];
+
+  chartCategories.forEach((c1) => {
+    const currentSeries: HeatMapSeries = {
+      name: c1,
+      data: [],
+    };
+    chartCategories.forEach((c2) => {
+      // the correlation of a company to itself is 1
+      if (c1 === c2) currentSeries.data.push(1);
+      else {
+        /* check if c1 + c2 are present as a pair in the mappedCorrelations ignoring order
+         if present push the corresponding correlation value, if no match is found place default value 0
+         it is mandatory that each series has the same amount and order of ordered values to display a meaningful x-Axis in the chart */
+        let valueToPush = 0;
+        mappedCorrelations.forEach((mc) => {
+          if (
+            JSON.stringify([c1, c2].sort()) ===
+            JSON.stringify(mc.slice(0, 2).sort())
+          )
+            // eslint-disable-next-line prefer-destructuring
+            valueToPush = mc[2];
```

Comment:
Is there a good reason to ignore the eslint rule or could this be done in another way?

Reply:
About the visual bug: Acknowledged but seems to be an Apex issue. The "reverseNegativeShade" property does also not work correctly (see TODO comment)

About the linter rule: I don't see how I should use Array destructuring here. I want to assign a new value to the existing let not create a new one.

Reply:
I think `[,,valueToPush] = mc` would work, but that'd be ugly.

The problem here is not the eslint rule, but the type of `mc` and how it's used. Why is `MappedCorrelation` a fixed-length array with different types (TypeScript calls it "tuple") and not an object?